keps/sig-api-machinery/2876-crd-validation-expression-language/README.md

```diff
@@ -327,17 +320,67 @@ like the `all` macro, e.g. `self.all(listItem, <predicate>)` or `self.all(mapKey
   all validation mechanisms (e.g. the OpenAPIV3 `maxItems` restriction), not just CEL validator
   rules. xref rule 4 in [specifying a structural schema](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#specifying-a-structural-schema).
 
-- We plan to allow access to the current state of the object to allow validation rules to check the
-  new value against the current value, e.g. for immutability checks (for validation racheting we would
-  prefer an approach like described in https://github.com/kubernetes/kubernetes/issues/94060 be pursued).
-
 - If the CEL evaluation exceeds the bounds we set (details below), the server will return a 408
   (Request Timeout) HTTP status code. The timeout will be a backstop we expect to rarely be used
   since CEL evaluations are multiple orders of magnitude faster that typical webhook invocations,
   and we can use CEL expression complexity estimations
   ([xref](https://github.com/jinmmin/cel-go/blob/a661c99f8e27676c70fc00f4f328476ca4dcdb7f/cel/program.go#L265))
   during CRD update to bound complexity.
 
+#### Transition Rules
+
+A rule that contains an expression referencing the identifier `oldSelf` is implicitly considered a
+"transition rule". Transition rules allow schema authors to prevent certain transitions between two
+otherwise valid states. For example:
+
+```yaml
+type: string
+enum: ["low", "medium", "high"]
+x-kubernetes-validations:
+- rule: "!(self == 'high' && oldSelf == 'low') && !(self == 'low' && oldSelf == 'high')"
+  message: cannot transition directly between 'low' and 'high'
+```
+
+Unlike other rules, transition rules apply only to operations meeting the following criteria:
+
+- Both an old and a new value exist. It remains possible to check if a value has been added or
```

Comment:
might be clearer to make "- Only run on update requests, never create requests" a first, standalone bullet